client/asset/eth/multirpc.go

```diff
@@ -985,6 +990,61 @@ func (m *multiRPCClient) withOne(ctx context.Context, providers []*provider, f f
 	return
 }
 
+func (m *multiRPCClient) withAll(ctx context.Context, f func(context.Context, *provider) error,
+	acceptabilityFilters ...acceptabilityFilter) error {
+	var atLeastOne bool
+	var errs []error
+	for _, p := range m.nonceProviderList() {
+		if p.failed() {
+			continue
+		}
+		ctx, cancel := context.WithTimeout(ctx, defaultRequestTimeout)
+		err := f(ctx, p)
+		cancel()
+		if err == nil {
+			atLeastOne = true
+			continue
+		}
+		var discarded bool
+		for i, f := range acceptabilityFilters {
+			discard, propagate, fail := f(err)
+			discarded = discard && (discarded || i == 0)
```

Comment:
I will revise this so `atLeastOne` means what it suggests, and that no error is returned from it.

`propagate` itself might be problematic in this method as well as it contradicts our intent with `atLeastOne`.

Reply:
Leaving as-is, with docs to clarify the function of `withAll`.  A nil error sets `atLeastOne=true`, causing `nil` to be returned to the caller of `withAll`, unless a subsequent provider hits an explicitly "propagated" error.  There are no such current uses, but it should be sensible for a future use case that has a propagateable error in mind.

On a related note, I've concluded that there is no valid use case for more than 1 filter, and the argument is variadic simply to allow 0 or 1.  I'm still running all filters if there are more than one, but it's not clear why a given error would match more than one filter or why it would even be required to define more than a single filter at all, hence my conclusion.